tools/c7n_gcp/c7n_gcp/resources/network.py

```diff
@@ -171,6 +212,39 @@ def get(client, resource_info):
                         'firewall': resource_info['resourceName'].rsplit('/', 1)[-1]})
 
 
+@Firewall.filter_registry.register('port-range')
+class PortRangeFirewallFilter(PortRangeFilter):
+    permissions = ('compute.firewalls.get', 'compute.firewalls.list')
+
+
+@Firewall.filter_registry.register('attached-to-cluster')
+class AttachedToClusterFirewallFilter(ValueFilter):
+    """
+    Checks if a firewall rule belongs to the network among the available clusters.
+
+    Usage example:
+      policies:
+       - name: gcp-firewall-attached-to-cluster-filter
+         resource: gcp.firewall
+         filters:
+         - attached-to-cluster
+    """
+    permissions = ('container.clusters.list',)
+
+    def process(self, resources, event=None):
+        session = local_session(self.manager.session_factory)
+        parent = 'projects/{}/locations/-'.format(session.get_default_project())
+        client = session.client('container', 'v1', 'projects.locations.clusters')
```

Comment:
use `self.manager.get_resource_manager('gke-cluster').resources()` to take advantage of caching.

Reply:
fixed